sheepdog/transactions/upload/__init__.py

```diff
@@ -32,8 +36,27 @@ def single_transaction_worker(transaction, *doc_args):
             transaction.flush()
             transaction.post_validate()
             transaction.commit()
-        except HandledIntegrityError:
-            pass
+        except IntegrityError:
+            transaction.session.rollback()
```

Comment:
`transaction.session` is just `session` here?

Reply:
I think they are referring to each other right? So it does not matter which one should be used

Reply:
Right, both are fine. (sidenote: I don't like the definition of `TransactionBase.session` at all ;) but anyway :P )